docs/DesignDocs/ExtendedEnclaveInitializationData_V2.md

```diff
@@ -0,0 +1,182 @@
+
+- Create a SHA256 context and restore the internal state to `hash_state`,
+- re-create the base image hash from the SHA256 context
+- measure the additional pages added after the base image. For EEID enclave with the size of heap/stack and #TCS specified at enclave signing time, the only additional page is the `oe_eeid_t` page. For EEID enclave with the size of heap/stack and #TCS selected by the enclave loader at enclave loading time, the additional pages include the heap and thread sections with stack using the `size_settings` in `oe_eeid_t`. Note that the number of TCS pages determines the number of thread sections. The measurement of the heap and thread section starts at address `vaddr`+4KB and includes `entry_point` in the TCS control pages. The measurement of the eeid page at `vaddr` is added last.
+- check that the final hash matches the hash reported in the extended report (e.g. MRENCLAVE),
+- check the identity of the signer of the extended image (e.g. public key corresponding to `OE_DEBUG_SIGN_KEY`),
+- check that the base image signature matches the ones in the extended report, except the image hash field (e.g. `sigstruct.EnclaveHash` in the case of SGX) and the signature section
+- check that the base image hash matches the value in the base image signature (e.g. `sigstruct.EnclaveHash` in the case of SGX), and finally
+- check the signature of the base image (e.g. `sigstruct` in the case of SGX; note this is integrity-protected by the hardware because of it's inclusion in the EEID pages).
+
+The claims produced by the enclave attestation verification should include config_id and config_svn, as well as the base enclave's identity information from the base enclave signature. The EEID enclave attestation verification process confirms that the enclave loader follows the SW convention to add the EEID page (in EEID enclave with loader selected heap/stack size and #TCS, also the heap and thread sections) after the base image, by checking the expected measurements. So the base enclave's identity information recorded in the EEID page can be trusted to reflect the base enclave loaded.
+
+As the enclave developer owns the definition of config_id and config_svn, the developer is responsible for providing the configuration data identified by config_id/config_svn to the Relying Party, if it's necessary for the Relying Party to analyze the configuration data. The Relaying Party should make sure the identity of the configuration data received matches config_id/config_svn claims produced by the enclave attestation verification process.
+
+For SGX KSS feature based solution, the attestation evidence verification implementation will be different, but the claims produced will also include config_id/config_svn as well as other enclave identity information, extracted from the HW generated enclave attestation data. The low level implementation difference between EEID based solution and SGX KSS feature based solution is not exposed to the Relaying Party.
```

Comment:
I am OK with the idea to use the same claim name implemented differently with or without KSS - however I want to make sure that we understand the impact on enclave runtime configuration. KSS configId is meant to be used for enclave-specific/application configuration so we need to make sure we have a clear plan on how we attest all runtime settings, including maximal heap, stack and TCS with EDMM. If it turns out that with KSS and EDMM we still have settings that can only be set at enclave signing time and that we want to change at enclave loading we may have to re-introduced a loader-specific hack on top of configId or mrenclave again...

Reply:
That's a good point @ad-l; if I understand correctly (and I may not), the memory settings are still fixed at signing time with EDMM, but now they allow a range of settings instead of one specific one. I think that may be fine for most applications, but there may be some that want tighter control than a large range of settings, e.g. for #TCS. @bodzhang: correct me if I'm wrong about that. Can you think of other new settings that we may want to change but still have attested?